src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,290 @@
+      return;
+    }
+    container.el.style.minWidth = this.options.suggestionWidth;
+    container.el.style.maxWidth = this.options.suggestionWidth;
+  }
+
+  private autoCreateComponentsInsideResult(element: HTMLElement, result: IQueryResult): IInitResult {
+    Assert.exists(element);
+    return Initialization.automaticallyCreateComponentsInsideResult(element, result);
+  }
+
+  private buildPreviewHeader(suggestion: string) {
+    const text = $$('span', {}, `${this.options.headerText} "${suggestion}"`).el;
+    const header = $$('div', { className: 'coveo-preview-header' }, text).el;
+    this.previewContainer.appendChild(header);
+  }
+
+  private buildResultsContainer() {
+    const resultsContainer = $$('div', { className: 'coveo-preview-results' }).el;
+    if (!this.options.previewWidth) {
+      return resultsContainer;
+    }
+
+    resultsContainer.style.width = `${this.options.previewWidth}px`;
+    return resultsContainer;
+  }
+
+  private get previewContainer() {
+    return $$(this.root).find('.coveo-preview-container');
   }
 
   private querySuggestGetFocus(args: IQuerySuggestSelection) {
-    if (!this.shouldShowPreviewResults) {
+    if (!this.previewContainer) {
+      return;
+    }
+
+    if (this.previousSuggestionHovered === args.suggestion) {
       return;
     }
-    this.executeQueryHover(args.suggestion);
+
+    this.previousSuggestionHovered = args.suggestion;
+    this.timer && clearTimeout(this.timer);
+    this.timer = setTimeout(() => {
+      this.executeQueryHover(args.suggestion);
+    }, this.options.hoverTime);
   }
 
   private executeQueryHover(suggestion: string) {
     const previousQueryOptions = this.queryController.getLastQuery();
     previousQueryOptions.q = suggestion;
     previousQueryOptions.numberOfResults = this.options.numberOfPreviewResults;
-    //TODO: I will need to execute a query, with the result,
-    //      build a container and display the result  next to the querySuggest
+    this.queryController
+      .getEndpoint()
+      .search(previousQueryOptions)
+      .then(results => {
+        $$(this.previewContainer).empty();
+        if (!results) {
+          return;
+        }
+        this.addImage(results);
+        this.buildPreviewHeader(suggestion);
+        this.buildResultsPreview(results);
+      });
+  }
+
+  private buildResultsPreview(results: IQueryResults): Promise<HTMLElement[]> {
+    const resultsContainer = this.buildResultsContainer();
+    this.previewContainer.appendChild(resultsContainer);
+    this.setupRenderer(resultsContainer);
+    this.buildResults(results).then(HTMLResult => {
+      if (!(HTMLResult.length > 0)) {
+        return;
+      }
+      this.updateResultPerRow(HTMLResult);
+      this.renderer.renderResults(HTMLResult, true, result => {});
+    });
+    return;
+  }
+
+  private updateResultPerRow(HTMLElements: HTMLElement[]) {
+    let size = HTMLElements.length == 4 ? 50 : Math.max(33, ~~(100 / HTMLElements.length));
```

Comment:
**General explication:**
This is to have a grid for displaying the results.
How it works is that the resultTemplate will render the htmlElements.
Depending on the number total of results, each element will receive X% of the space on the row to render itself.
if we have 
```
1 result => take 100% of the space on the row => 1 item on the first row
2 result => take 50% of the space on the row => 2 item on the first row
3 result => take 33% of the space on the row => 3 item on the first row
4 result => take 50% of the space on the row  => talked to UX and we want a 2x2 grid
5 result => take 33% of the space on the row => a 3x2 grid
6 result => take 33% of the space on the row1  => a 3x3 grid
```


**What the number means:**
- 33: We want a maximum of 3 items per row so each item, will receive a minimum of `33%` of the space on the row
- 4: if we have 4 elements to display, we want a 2x2 grid, not a 3x1 so each item will receive 50% of the space on the row

**Division by 0:**
0 will not be an issue since I checked before entering the function at line 265. Also, this is a query from a querySuggest so I would find it weird if our querySuggest would return 0 result.